common/cmake/v2x.cmake

```diff
@@ -0,0 +1,232 @@
+    endif()
+
+    append_file_dependency(DEPENDS_LIST ${SRC})
+    append_file_includes(INCLUDES ${SRC})
+
+    get_file_target(SRC_TARGET ${SRC})
+    get_target_property(INCLUDE_FILES ${SRC_TARGET} INCLUDE_FILES)
+    foreach(INCLUDE_SRC ${INCLUDE_FILES})
+      get_filename_component(INCLUDE_SRC_DIR ${INCLUDE_SRC} DIRECTORY)
+      get_filename_component(INCLUDE_ROOT ${INCLUDE_SRC} NAME_WE)
+
+      append_file_dependency(DEPENDS_LIST ${INCLUDE_SRC_DIR}/${INCLUDE_ROOT}.model.xml)
+      append_file_dependency(DEPENDS_LIST ${INCLUDE_SRC_DIR}/${INCLUDE_ROOT}.pb_type.xml)
+      list(APPEND MODEL_INCLUDE_FILES ${INCLUDE_SRC_DIR}/${INCLUDE_ROOT}.model.xml)
+      list(APPEND PB_TYPE_INCLUDE_FILES ${INCLUDE_SRC_DIR}/${INCLUDE_ROOT}.pb_type.xml)
+    endforeach()
+  endforeach()
+
+  list(GET V2X_SRCS 0 FIRST_SOURCE_FILE)
+  get_file_location(FIRST_SOURCE ${FIRST_SOURCE_FILE})
+
+  set(TOP_ARG "")
+  if(NOT ${V2X_TOP_MODULE} STREQUAL "")
+    set(TOP_ARG "--top=${V2X_TOP_MODULE}")
+  endif()
+
+  string(
+    REPLACE
+      ";"
+      ","
+      INCLUDES_LIST
+      "${INCLUDES}"
+  )
+
+  set(INCLUDE_ARG "")
+  if(NOT "${INCLUDES_LIST}" STREQUAL "")
+    set(INCLUDE_ARG "--includes=${INCLUDES_LIST}")
+  endif()
+
+  add_custom_command(
+    OUTPUT "${V2X_NAME}.pb_type.xml"
+    DEPENDS
+      ${DEPENDS_LIST}
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_pbtype.py
+    COMMAND
+      ${CMAKE_COMMAND} -E env YOSYS=${YOSYS}  ${PYTHON3} ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_pbtype.py ${TOP_ARG}
+      -o ${CMAKE_CURRENT_BINARY_DIR}/${V2X_NAME}.pb_type.xml ${FIRST_SOURCE}
+      ${INCLUDE_ARG}
+    WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/
+  )
+  add_file_target(FILE "${V2X_NAME}.pb_type.xml" GENERATED)
+  get_file_target(SRC_TARGET_NAME "${V2X_NAME}.pb_type.xml")
+  set_target_properties(${SRC_TARGET_NAME} PROPERTIES INCLUDE_FILES "${PB_TYPE_INCLUDE_FILES}")
+
+  add_custom_command(
+    OUTPUT "${V2X_NAME}.model.xml"
+    DEPENDS
+      ${DEPENDS_LIST}
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_model.py
+    COMMAND
+      ${CMAKE_COMMAND} -E env YOSYS=${YOSYS}  ${PYTHON3} ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_model.py ${TOP_ARG}
+      -o ${CMAKE_CURRENT_BINARY_DIR}/${V2X_NAME}.model.xml ${FIRST_SOURCE}
+      ${INCLUDE_ARG}
+    WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/
+  )
+  add_file_target(FILE "${V2X_NAME}.model.xml" GENERATED)
+  get_file_target(SRC_TARGET_NAME "${V2X_NAME}.model.xml")
+  set_target_properties(${SRC_TARGET_NAME} PROPERTIES INCLUDE_FILES "${MODEL_INCLUDE_FILES}")
+
+  add_custom_target(
+    ${V2X_NAME}
+    DEPENDS
+        "${V2X_NAME}.model.xml"
+        "${V2X_NAME}.pb_type.xml"
+  )
+
+endfunction(V2X)
+
+function(VPR_TEST_PBTYPE)
+  # ~~~
+  # VPR_TEST_PBTYPE(NAME <name>)
+  # ~~~
+  #
+  # Run the pb_type.xml file through vpr to check it is valid.
+  set(options)
+  set(oneValueArgs NAME)
```

Comment:
NAME should be the output target name.  You should have two arguments, one for the output target name, and one for the pb_type name.  This enables sufficient output name control to avoid collisions when required.

Reply:
Done.